api/search.go

```diff
@@ -193,78 +173,225 @@ func CreateRequests(w http.ResponseWriter, req *http.Request, validator QueryPar
 		return "", nil, nil
 	}
 
-	// create SearchRequest with all the compulsory values
+	datasetIDs, datasetErr := parseDatasetIDs(ctx, params)
+	if datasetErr != "" {
+		http.Error(w, datasetErr, http.StatusBadRequest)
+		return "", nil, nil
+	}
+
+	// Create SearchRequest
+	reqSearch := createSearchRequest(sanitisedQuery, offset, limit, contentTypes, fromDate.(query.Date), toDate.(query.Date), topics, sort, highlight, datasetIDs, nlpCriteria)
+
+	// Process additional parameters like Population Types, Dimensions, and Dataset IDs
+	reqSearch.PopulationTypes = parsePopulationTypes(params)
+	reqSearch.Dimensions = parseDimensions(params)
+
+	// Create CountRequest
+	reqCount := createCountRequest(sanitisedQuery)
+
+	if cfg.DebugMode {
+		log.Info(ctx, "[DEBUG]", log.Data{"search_request": reqSearch})
+	}
+
+	return params.Get(ParamQ), reqSearch, reqCount
+}
+
+func sanitizeAndValidateQuery(ctx context.Context, w http.ResponseWriter, params url.Values) (string, error) {
+	q := params.Get(ParamQ)
+	sanitisedQuery := sanitiseDoubleQuotes(q)
+	queryHasSpecialChars := checkForSpecialCharacters(sanitisedQuery)
+
+	if queryHasSpecialChars {
+		log.Info(ctx, "rejecting query as it contains special characters", log.Data{"query": sanitisedQuery})
+		http.Error(w, "Invalid characters in query", http.StatusBadRequest)
```

Comment:
I don't think this func - a validator - should be calling `http.Error` - perhaps move this back to the caller